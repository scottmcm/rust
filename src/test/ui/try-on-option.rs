#![feature(try_trait)]

fn main() {}

fn foo() -> Result<u32, ()> {
    let x: Option<u32> = None;
    x?; //~ mismatched types
    Ok(22)
}

fn bar() -> u32 {
    let x: Option<u32> = None;
    x?; //~ mismatched types
    22
}
