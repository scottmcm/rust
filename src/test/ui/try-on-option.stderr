error[E0308]: mismatched types
  --> $DIR/try-on-option.rs:7:5
   |
LL |     x?; //~ mismatched types
   |     ^^ expected enum `std::result::Result`, found enum `std::option::Option`
   |
   = note: expected type `std::result::Result<u32, ()>`
              found type `std::option::Option<_>`

error[E0308]: mismatched types
  --> $DIR/try-on-option.rs:13:5
   |
LL |     x?; //~ mismatched types
   |     ^^ expected u32, found enum `std::option::Option`
   |
   = note: expected type `u32`
              found type `std::option::Option<_>`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
